specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,278 @@
+@summary("Get properties for existing schema.")
+@doc("Gets the properties referencing an existing schema within the specified schema group, as matched by schema content comparison.")
+@action("get-id")
+op getSchemaPropertiesByContent is resourceOperations.ResourceAction<
+  SchemasName,
+  {
+    ...SchemaContentType;
+
+    @doc("String representation (UTF-8) of the registered schema.")
+    @body
+    schemaContent: bytes;
+  },
+  SchemaProperties
+>;
+
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{schemaName}?api-version={apiVersion}
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Non-standard operation."
+#suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+@summary("Register new schema")
+@doc("Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.")
+@route("/$schemaGroups/{groupName}/schemas/{schemaName}")
+@put
+op registerSchema is Azure.Core.Foundations.Operation<
+  {
+    @doc("Name of schema group.")
+    @path
+    groupName: string;
+
+    @doc("Name of schema.")
+    @path
+    schemaName: string;
+
+    ...SchemaContent;
+    ...SchemaContentType;
+  },
+  NoContentResponse<SchemaProperties>
+>;
+
+// Models
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+union SchemaContentTypeValues {
+  avro: "application/json; serialization=Avro",
+
+  @added(ServiceVersion.V2022_10)
+  json: "application/json; serialization=Json",
+
+  @added(ServiceVersion.V2022_10)
+  custom: "text/plain; charset=utf-8",
+
+  @added(ServiceVersion.V2023_07_01)
+  protobuf: "text/vnd.ms.protobuf",
+}
+
+alias SchemaContentType = {
+  @doc("The content type for given schema.")
+  @header("Content-Type")
+  contentType: SchemaContentTypeValues;
+};
+
+alias SchemaContent = {
+  @body
+  @doc("String representation (UTF-8) of the schema.")
+  content: bytes;
+};
+
+@doc("Metadata of a schema.")
+model SchemaProperties {
+  @doc("URL location of schema, identified by schema group, schema name, and version.")
+  @header("Location")
+  location: string;
+
+  @doc("References specific schema in registry namespace.")
+  @header("Schema-Id")
+  schemaId: string;
+
+  @doc("URL location of schema, identified by schema ID.")
+  @header("Schema-Id-Location")
+  schemaIdLocation: string;
+
+  @doc("References schema group.")
+  @header("Schema-Group-Name")
+  schemaGroupName: string;
+
+  @doc("References schema name.")
+  @header("Schema-Name")
+  schemaName: string;
+
+  @doc("Version of the returned schema.")
+  @header("Schema-Version")
+  schemaVersion: int32;
+}
+
+@doc("The schema, including its metadata and content.")
+model Schema {
+  ...SchemaProperties;
+  ...SchemaContent;
+  ...SchemaContentType;
+  ...OkResponse;
```

Comment:
Is the status code/`OkResponse` really part of the conceptual model of a Schema? I would suggest that it is not and should be intersected when used as a response model in an operation.

Reply:
it is not, updating